beacon-chain/blockchain/process_block.go

```diff
@@ -125,6 +125,19 @@ func (s *Service) onBlock(ctx context.Context, signed *ethpb.SignedBeaconBlock,
 			return err
 		}
 	}
+	if featureconfig.Get().UpdateHeadTimely {
```

Comment:
We probably need to shift this even further above, as updating justified checkpoints can incur a cost in regenerating our state. Also we should send out a block received event after, so as to attest as quickly as possible

Reply:
Forkchoice requires updated justified checkpoint as base point